### PR TITLE
Compile Server and Client API into Libs

### LINKED ISSUE #6: Create Connect API

### PR DESCRIPTION
**Pull request by fsc1118**
I used [https://github.com/alvinkwok1/RDMA-EXAMPLE/](url) as base point to wrap RDMA APIs.

This will compile two static libraries  `librdma_client_lib.a` and `_librdma_server_lib.a`

`librdma_client_lib.a`  should have APIs database use to interact with memory pool. This library should be put  on GroundDB's compute nodes.

`librdma_server_lib.a` should have APIs memory pool does miscellaneous tasks such as to initialize itself. This library should be put on GroundDB's memory pools.

This commit addresses `rdma_connect`. The above two library has respectively `init_server` (used by server) and `connect` (used by client) to establish RDMA connection. The return value of those two functions contain important information for following RDMA operations.

### CODE REVIEW

File: GroundDBMemoryPool/src/include/memoryPool.hh
Comment (fsc1118) [inline GroundDBMemoryPool/src/include/memoryPool.hh:1]: I will put those RDMA definition in rdma.hh

File: GroundDBMemoryPool/src/include/rdma.hh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+{
+    uint64_t addr;   /* Buffer address */
+    uint32_t rkey;   /* Remote key */
+    uint32_t qp_num; /* QP number */
+    uint16_t lid;    /* LID of the IB port */
+    uint8_t gid[16]; /* gid */
+} __attribute__((packed));
+
+/* structure of system resources */
+struct resources
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/include/rdma.hh:45]: I believe `struct resources` suffices for following one side RDMA operation.

File: GroundDBMemoryPool/src/include/rdma_client.hh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#pragma once
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/include/rdma_client.hh:1]: Only  `connect` now. Should add more APIs in following commits

File: GroundDBMemoryPool/src/include/rdma_server.hh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+struct resources *init_server(
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/include/rdma_server.hh:3]: Only `init_server` now. Should add more in following commits

File: GroundDBMemoryPool/src/main_client.cc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#include <memoryPool.hh>
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/main_client.cc:1]: Example. Should delete in release

File: GroundDBMemoryPool/src/main_server.cc [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#include <memoryPool.hh>
+int main()
+{
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/main_server.cc:3]: Example. Should delete in release

File: GroundDBMemoryPool/src/rdma/rdma_client/connect.cc [hunk 6]
```diff
@@ -0,0 +1,40 @@
+#include <rdma.hh>
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/rdma/rdma_client/connect.cc:1]: `connect` basically is a wrap up of `connect` in [https://github.com/alvinkwok1/RDMA-EXAMPLE/](url)

File: GroundDBMemoryPool/src/rdma/rdma_server/init_server.cc [hunk 7]
```diff
@@ -0,0 +1,38 @@
+#include <rdma.hh>
```
Comment (fsc1118) [inline GroundDBMemoryPool/src/rdma/rdma_server/init_server.cc:1]: Similarly, this is a wrap up.

File: GroundDBMemoryPool/src/rdma/util.cc
Comment (fsc1118) [inline GroundDBMemoryPool/src/rdma/util.cc:1]: This is copied from [https://github.com/alvinkwok1/RDMA-EXAMPLE/](url)

File: GroundDBMemoryPool/src/rdma/util.h
Comment (fsc1118) [inline GroundDBMemoryPool/src/rdma/util.h:1]: This is copied from [https://github.com/alvinkwok1/RDMA-EXAMPLE/](https://github.com/GroundDB/GroundDB/pull/10/commits/url)

**Review by te-guo — APPROVED** [review]
We need to update the structs to support multiple connections & multiple memory regions in the future.

**Pull request merged**